ingestion/src/metadata/ingestion/source/database/datalake/metadata.py

```diff
@@ -565,6 +621,25 @@ def get_s3_files(client, key, bucket_name, client_kwargs=None):
             )
         return None
 
+    @staticmethod
+    def fetch_col_types(data_frame, column_name):
+        data_type = DATALAKE_DATA_TYPES.get(
+            data_frame[column_name].dtypes.name, DataType.STRING.value
+        )
+        if data_type == DataType.FLOAT.value:
+            try:
+                if data_frame[column_name].dropna().any():
+                    if isinstance(data_frame[column_name].iloc[0], dict):
+                        return DataType.JSON.value
+                    if isinstance(data_frame[column_name].iloc[0], str):
+                        return DataType.STRING.value
+            except Exception as err:
+                logger.warning(
+                    f"Failed to disinguish data type for column {column_name}, Falling back to {data_type}, exc: {err}"
+                )
+                logger.debug(traceback.format_exc())
+        return data_type
+
```

Comment:
Could you add some test around this and add some comment explaining what we are doing?

Reply:
yes, let me add that